### ISSUE TITLE
Graph Tree Mode not displaying contained devices

### ISSUE THREAD

**Issue by dsclassen** [Cacti/cacti#1169]
I'm on https://github.com/Cacti/cacti/commit/e65c26d3410022a9485902d15f97bfba725136fa
In the Graph tab in Tree Mode the various devices and/or graphs under the trees items are not displaying. I'm not sure when this bug showed up.
![treemodebug](https://user-images.githubusercontent.com/8882563/34538010-8025a3f8-f07f-11e7-81cc-e4a0699c935d.png)

**Comment by netniV** [comment]
There have been a number of topics on here that cover similar issues, have you checked through those first to see if any covers your issue?

**Comment by netniV** [comment]
Also, what version of cacti are you running?

**Comment by dsclassen** [comment]
I'm on the develop branch which is currently 1.1.29
I'll browse the other issues to see if something relevant pops up.
Thanks

**Comment by dsclassen** [comment]
maybe these errors showing up in my cacti.log file are a clue?

```
2018/01/03 15:30:58 - DBCALL ERROR: SQL Assoc Failed!, Error: Unknown column 'gti.site_id' in 'field list'
--
2018/01/03 15:30:58 - DBCALL ERROR: SQL Assoc Failed!, Error:1054, SQL:'SELECT gti.graph_tree_id AS tree_id, gti.id, gti.title, gti.host_id, gti.site_id, gti.local_graph_id, gti.host_grouping_type, h.description AS hostname, s.name AS sitename FROM graph_tree_items AS gti INNER JOIN graph_tree AS gt ON gt.id = gti.graph_tree_id LEFT JOIN host AS h ON h.id = gti.host_id LEFT JOIN sites AS s ON gti.site_id=s.id WHERE gti.local_graph_id=0 AND gti.parent=0 AND gti.graph_tree_id=6 AND gt.id IN (1, 2, 3, 6, 7, 8, 9, 10, 11, 12) ORDER BY gti.position'

```

**Comment by cigamit** [comment]
Hmm, that looks more like your gti is not up-to-date.  Odd.

**Comment by cigamit** [comment]
You must have missed some step in the upgrade process.

**Comment by dsclassen** [comment]
I just chug along on the develop branch, git pulling now and again. I feel like this happens to me every once and awhile. Somehow I miss a DB update or something. There has to be a way to have cacti check the database structure. 

Maybe whenever someone makes changes to cacti.sql an entry could be added to some database_update table, and then this table could be consulted to see at what stage of update your DB is. Does that make any sense?

**Comment by pautiina** [comment]
I had a similar problem.
Look for this https://github.com/Cacti/cacti/issues/1131 . Here you can find, how update missing table and database.
I think it's double for #1131 and can be close.

**Comment by netniV** [comment]
I have suspected before that occasionally the upgrade of the database does not happen for some reason.  My own DB didn't contain all the columns expected though I'm not entirely sure why there have been these few instances of upgrading that fails to update the database.  

The issue #1131 that @pautiina mentions had the same issue too where the upgrade did not appear to happen.

I may run through a VM through the various releases jumping around to see if I can get the upgrade to fail and why.

As for your suggestion about the DB upgrade, the process is this:

1. Check the CACTI_VERSION file matches the currently running system. 
2. If it is newer, redirect the user to /install/
3. Run through the installation checks to make sure all dependencies etc match
4. Update the DB and write the new version

The scripts in the /install/upgrades/ folder are used to perform any upgrades and are based off the version number.  site_id was in 1.1.4 which was the same issue we discovered fir @pautiina
